FirebaseSessions/Sources/SettingsDownloadClient.swift

```diff
@@ -0,0 +1,99 @@
+//
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+@_implementationOnly import GoogleUtilities
+
+protocol SettingsDownloadClient {
+  func fetch(completion: @escaping (Result<[String: Any], Error>) -> Void)
+}
+
+class SettingsDownloader: SettingsDownloadClient {
+  private let appInfo: ApplicationInfoProtocol
+  private let identifiers: IdentifierProvider
+  private let installations: InstallationsProtocol
+
+  init(appInfo: ApplicationInfoProtocol, identifiers: IdentifierProvider,
+       installations: InstallationsProtocol) {
+    self.appInfo = appInfo
+    self.identifiers = identifiers
+    self.installations = installations
+  }
+
+  func fetch(completion: @escaping (Result<[String: Any], Error>) -> Void) {
+    guard let url = url else {
```

Comment:
For readability purposes, I might recommend not shadowing the `url` private getter with the url variable. Maybe name the variable `validURL`?

Reply:
Done